### PR TITLE
2.0

### PR DESCRIPTION
**Pull request by bennypowers**
There's one test still failing: if the element has `required` set, then it will validate as soon as it's instanciated.

### CODE REVIEW

File: date-input.html [hunk 1]
```diff
@@ -240,7 +251,8 @@
     },
 
     _selectYear: function() {
-      this.$.expirationYear.focus();
+      var yearInput = this.$.nativeYearInput || this.$.expirationYear;
+          yearInput.focus();
```
Comment (gurpreetatwal) [inline date-input.html]: /nit this should be aligned with the var

File: date-input.html [hunk 2]
```diff
@@ -265,6 +277,21 @@
...
+    },
+
+    _onBlur: function(e) {
+      var container = this.domHost.$.container;
+          container._boundOnBlur(e);
+    },
+
+    _onInput: function(e) {
+      var container = this.domHost.$.container;
+          container._boundOnInput(e);
```
Comment (gurpreetatwal) [inline date-input.html]: What's the context for this change? Genuinely curious as to what these do/are for.

File: gold-cc-expiration-input.html [hunk 3]
```diff
@@ -80,11 +81,13 @@
           readonly$="[[readonly]]">
       </date-input>
 
-      <template is="dom-if" if="[[errorMessage]]">
-        <paper-input-error slot="add-on" id="error">
-          [[errorMessage]]
-        </paper-input-error>
-      </template>
+      <dom-if if="[[errorMessage]]">
```
Comment (gurpreetatwal) [inline gold-cc-expiration-input.html]: why does this need to be wrapped again? afaik the other original syntax is compatible with both versions of Polymer

File: test/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (gurpreetatwal) [inline test/.eslintrc.json]: same as other comment

File: test/basic.html [hunk 5]
```diff
@@ -39,9 +39,13 @@
 
     suite('basic', function() {
       test('invalid input is not ok', function() {
+        console.log('before fix')
         var input = fixture('basic');
-        input.value='1234';
+        console.log('after fix')
```
Comment (gurpreetatwal) [inline test/basic.html]: stray `console.log`s

File: test/basic.html [hunk 6]
```diff
@@ -39,9 +39,13 @@
...
         forceXIfStamp(input);
+        console.log('after stamp')
+        input.value='1234';
+        console.log('after val')
```
Comment (gurpreetatwal) [inline test/basic.html]: this is straight up pattern matching, but do you think the tests need to be wrapped in `Polymer.Base.async`?

That was done on the tests for [`gold-cc-input`](https://github.com/PolymerElements/gold-cc-input/pull/60/files#diff-687a8d8dd06bbcfea27dd116b48318c3R69)

**Comment by gurpreetatwal** [comment]
Also I like the [helper function](https://github.com/PolymerElements/gold-cc-input/pull/60/files#diff-687a8d8dd06bbcfea27dd116b48318c3R55) that was created for the `gold-cc-input` tests. That way the tests themselves don't really have to worry about getting the right input, they just use the abstraction

**Comment by gurpreetatwal** [comment]
As for the `required` test failing, we just need to port over [this section of logic](https://github.com/PolymerElements/gold-cc-input/pull/60/files#diff-025b7cfad2c6915fb0dbf37ff621c5f3R243)

File: test/basic.html (hunk 6 above)
Comment (bennypowers) [reply test/basic.html]: Could help, not sure

File: gold-cc-expiration-input.html (hunk 3 above)
Comment (bennypowers) [reply gold-cc-expiration-input.html]: Can't remember :D If it works...

File: .eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply .eslintrc.json]: Up to you, I find it helps my development

File: date-input.html (hunk 2 above)
Comment (bennypowers) [reply date-input.html]: Honestly can't remember :D

**Comment by bennypowers** [comment]
I've allowed edits from maintainers. Sorry to toss the ball back to you, but I'm stacked at work ATM.

**Comment by gurpreetatwal** [comment]
No worries, I'm currently also busy with school and work 😅 . I'll try to play with the code if I get some free time.

**Comment by gurpreetatwal** [comment]
Hey @bennypowers I don't actually use Polymer anymore, I'm going to be deleting this fork. You should just fork the upstream and make a PR to them

**Pull request closed without merging**